stdlib/public/core/Optional.swift

```diff
@@ -328,6 +394,40 @@ public struct _OptionalNilComparisonType : ExpressibleByNilLiteral {
   public init(nilLiteral: ()) {
   }
 }
+
+/// Returns a Boolean value indicating whether an argument matches `nil`.
+///
+/// You can use the pattern-matching operator (`~=`) to test whether an
+/// optional instance is `nil` even when the wrapped value's type does not
+/// conform to the `Equatable` protocol. The following example declares the
+/// `numbers` variable as an optional array of integers. Although `Array<Int>`
+/// is not an `Equatable` type, this operator allows matching against `nil`.
+///
+///     var numbers: [Int]?
+///     if nil ~= numbers {
```

Comment:
Do we want to encourage people to do this? I think of this `~=` as kind of an implementation detail of the `switch` usage (esp. given it takes an `_` protocol as its arg).